tex2wiki/src/tex2wiki.py

```diff
@@ -0,0 +1,618 @@
+
+    return result
+
+
+def generate_math_html(text, options=None, spacing=2):
+    # type: (str(, dict, int)) -> str
+    """Special case of generate_html, where the tag is "math"."""
+    if options is None:
+        options = {}
+
+    option_text = [key + "=\"" + value + "\"" for key, value in options.iteritems()]
+    result = "<math" + " " * (option_text != []) + ", ".join(option_text) + ">"
+
+    if spacing == 2:
+        result += "{\\displaystyle \n" + text + "\n}</math>\n"
+    elif spacing == 1:
+        result += "{\\displaystyle " + text + "}</math>\n"
+    else:
+        result += "{\\displaystyle " + text + "}</math>"
+
+    return result
+
+
+def generate_link(left, right=""):
+    # type: (str(, str)) -> str
+    """Generates a MediaWiki link."""
+    if right == "":
+        return "[[" + left + "|" + left + "]]"
+
+    return "[[" + left + "|" + right + "]]"
+
+
+def multi_split(s, seps):
+    # type: (str, list) -> list
+    """Splits a string on multiple characters."""
+    res = [s]
+    for sep in seps:
+        s, res = res, []
+        for seq in s:
+            res += seq.split(sep)
+    return res
+
+
+def convert_dollar_signs(string):
+    # type: (str) -> str
+    """Converts dollar signs to html for math mode."""
+    count = 0
+    result = ""
+    for i, ch in enumerate(string):
+        if ch == "$":
+            if count % 2 == 0:
+                result += "<math>{\\displaystyle "
+            else:
+                result += "}</math>"
+            count += 1
+        else:
+            result += ch
+
+    return result
+
+
+def format_formula(formula):
+    """Obtain the raw formula (integers), as well as the formatted version."""
+    if formula[0] == ":":  # handling for Jacobi special stuff
+        return [str(int(formula[1:]))], formula[1:].zfill(2)
+
+    formula = multi_split(formula.split("Formula:", 1)[1], [".", ":"])
+
+    for j in [-1, -2, -3]:
+        formula[j] = formula[j].zfill(2)
+
+    raw = copy.deepcopy(formula)
+
+    # remove any text from the raw equation
+    i = 0
+    while i < len(raw):
+        if not raw[i].isdigit():
+            raw.pop(i)
+        else:
+            raw[i] = str(int(raw[i]))  # removes any 0s that are present from left end.
+            i += 1
+
+    if len(formula) == 3:
+        formula = formula[0] + "." + formula[1] + ":" + formula[2]
+    else:
+        formula = ":".join(formula[:-3]) + ":" + formula[-3] + "." + formula[-2] + ":" + formula[-1]
+
+    return raw, formula
+
+
+def format_metadata(string):
+    # type: (str) -> str
+    """Formats the metadata of an equation."""
+    if string == "":
+        return ""
+
+    # modified convert_dollar_sign algorithm
+    dollar_count = 0
+    result = ""
+    for i, ch in enumerate(string):
+        if ch == "$":
+            if dollar_count % 2 == 1:
+                result += "}</math>"
+            elif dollar_count > 1:
+                result = result[:-1] + "<br />\n<math>{\\displaystyle "
```

Comment:
no br no displaystyle please, you can use the option dispay=block in the math tag if you whish